File: hdlibrary/src/main/java/com/hd/base/BaseViewModel.java
package com.hd.base;

import android.content.Context;
import android.databinding.BaseObservable;
import android.databinding.ObservableBoolean;
import android.databinding.ObservableField;

import com.titan.data.source.DataRepository;

/**
 * Created by hanyw on 2017/12/4/004
 */

public class BaseViewModel extends BaseObservable {
    protected Context mContext;
    protected DataRepository mDataRepository;
    //数据加载中
    public final ObservableBoolean dataLoading = new ObservableBoolean(false);
    //提示信息
    public final ObservableField<String> snackbarText = new ObservableField<>();
    public String getSnackbarText() {
        return snackbarText.get();
    }

}

File: app/src/main/java/com/hd/share/BaseViewModel.kt
package com.hd.share

import android.content.Context
import android.databinding.BaseObservable
import android.databinding.ObservableBoolean
import android.databinding.ObservableField
import com.hd.data.DataRepository


/**
 * Created by whs on 2017/5/5
 */

class BaseViewModel : BaseObservable() {

    protected var mContext: Context? = null
    protected var mDataRepository: DataRepository? = null
    //数据加载中
    val dataLoading = ObservableBoolean(false)
    //提示信息
    val snackbarText: ObservableField<String> = ObservableField()

    fun getSnackbarText(): String {
        return snackbarText.get()
    }

}
